src/hotspot/share/classfile/javaClasses.cpp

```diff
@@ -1729,6 +1730,20 @@ void java_lang_Thread::set_jvmti_thread_state(oop java_thread, JvmtiThreadState*
   java_thread->address_field_put(_jvmti_thread_state_offset, (address)state);
 }
 
+int java_lang_Thread::VTMS_transition_disable_count(oop java_thread) {
+  return java_thread->int_field(_jvmti_VTMS_transition_disable_count_offset);
+}
+
+void java_lang_Thread::inc_VTMS_transition_disable_count(oop java_thread) {
+  int val = VTMS_transition_disable_count(java_thread);
+  java_thread->int_field_put(_jvmti_VTMS_transition_disable_count_offset, val + 1);
+}
+
+void java_lang_Thread::dec_VTMS_transition_disable_count(oop java_thread) {
+  int val = VTMS_transition_disable_count(java_thread);
+  java_thread->int_field_put(_jvmti_VTMS_transition_disable_count_offset, val - 1);
+}
```

Comment:
These are not thread-safe functions. What constraints exist for using these methods safely?
Update: looks like they must be called with the lock held so we should assert that.
Should also assert the count never goes negative (which I assume would be an error?).

Reply:
Each of these two functions is called just once with the lock held.
I feel that asserting this would be an overkill but I can add it if you still think it is important.
I've added assert for negative count.

Reply:
Done.